---
title: "Salmon_Quantification"
author: 
  - "Bioinformatics Resource Center"
  - "Rockefeller University"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Salmon Quantification
 This code describes the methods used to quantify transcipts using the pseudoaligner Salmon. 

## Install Salmon
We use the Herper package to install salmon into a miniconda environment. 

```{r}
library(Herper)
salmon_paths <- install_CondaTools(tools = "salmon", env = "RNAseq_analysis")
salmon_paths

```

## Build Salmon Index
We will build the salmon index using the reference information from Bioconductor packages: BSgenome and TxDb.

```{r}
library(BSgenome.Mmusculus.UCSC.mm10)
library(TxDb.Mmusculus.UCSC.mm10.knownGene)

allTxSeq <- extractTranscriptSeqs(BSgenome.Mmusculus.UCSC.mm10, TxDb.Mmusculus.UCSC.mm10.knownGene,
    use.names = TRUE)
writeXStringSet(allTxSeq, "mm10Trans.fa")

fastaTx <- "mm10Trans.fa"
indexName <- "mm10Trans"
with_CondaEnv("RNAseq_analysis",
                        system2(command="salmon",
                        args = c("index",
                        "-i", indexName,
                        "-t", fastaTx),
                        stdout = TRUE))

```

## Quantify transcipt abundance
Finally we will use the salmon *quant* command to quantify the transcript abundance for our every transcript. 

```{r}

fq <- "GSM5291545_CB_CreNeg_input_R2.fastq.gz"
outDir <- "GSM5291545_CB_CreNeg_input_R2"

with_CondaEnv("RNAseq_analysis",
                      system2(command="salmon", args = c("quant",
                        "-i", indexName,
                        "-o", outDir,
                        "-l A",
                        "-r", fq),
                        stdout = TRUE))

```
